pkg/kds/global/components_test.go

```diff
@@ -167,4 +169,33 @@ var _ = Describe("Global Sync", func() {
 		}, "3s", "100ms").Should(Equal(2))
 	})
 
+	It("should have up to date list of provided types", func() {
+		excludeTypes := map[model.ResourceType]bool{
+			mesh.DataplaneInsightType:  true,
+			mesh.DataplaneOverviewType: true,
+			mesh.ServiceInsightType:    true,
+			mesh.ServiceOverviewType:   true,
+		}
+
+		// take all mesh-scoped types and exclude types that won't be synced
+		actualProvidedTypes := []model.ResourceType{}
+		for _, typ := range registry.Global().ListTypes() {
+			obj, err := registry.Global().NewObject(typ)
+			Expect(err).ToNot(HaveOccurred())
+			if obj.Scope() == model.ScopeMesh && !excludeTypes[typ] {
+				actualProvidedTypes = append(actualProvidedTypes, typ)
+			}
+		}
+
+		// plus 2 global-scope types
+		extraTypes := []model.ResourceType{
+			mesh.MeshType,
+			system.ConfigType,
+		}
+
+		actualProvidedTypes = append(actualProvidedTypes, extraTypes...)
+
+		Expect(actualProvidedTypes).To(ConsistOf(global.ProvidedTypes))
```

Comment:
one nit: I'd also check that `len(actualProvidedTypes) == len(global.PRovidedTypes)`

Reply:
`ConsistOf` tracks this as well, but don't mind saving some CPU loops and do fast fail with this length check, fixed

Reply:
Oh, I'm sorry! I thought that when `actualProvidedTypes` has one extra element then this check will pass.